src/pass/llvmgen.cpp

```diff
@@ -37,7 +40,7 @@ Value* LLVMGen::llcall(const string name, llvm::Type* ret_type,
     return llcall(name, ret_type, arg_vals);
 }
 
-llvm::Type* LLVMGen::lltype(const DataType& type)
+llvm::Type* LLVMGen::lltype(const DataType& type, unsigned int addr_space)
```

Comment:
For the alloc address space issue, it doesn't work to indiscriminately set it to 5 (local) within lltype, because this messes with other instructions that won't necessarily be working on local memory (eg, load instruction operating on a global array). I've instead added addr_space as a parameter that defaults to 0 (generic). Then I set it to 5 when create the alloc instruction. Let me know if you prefer it to be handled differently.